### DISCUSSION TITLE
Hidden Input value not read by AlpineJS

### DISCUSSION THREAD

**Discussion by mdmoura** [alpinejs/alpine#3183] [7. Bugs (archived)] [answered]
I have the following form using AlpineJS:

    <form x-data="inquiry()" x-on:submit.prevent="submit" method="post">
      <label>Email</label>
      <input type="text" name="email" x-model="data.email">
      <input type="hidden" name="subject" value="my subject" x-model="data.subject">
      <button>Submit</button>
    </form>

And the submission javascript code is:

    <script>
      function inquiry() {
        return {
          data: {
            email: "",
            subject: ""
          },
          submit() {
            console.log(JSON.stringify(this.data));
          }
        };
      }
  </script>

When filling the form's email and submitting it I get the email's value in the console.

However the subject value from the hidden input is always empty.

Doesn't AlpineJS supports hidden inputs? Am I missing something or this is a bug?

**Comment by SimoTod** [comment]
Why don't you put the value in x-data. X-model overrides the value attribute with whatever is in your data so your hidden input gets wiped out when Alpine starts

**Comment by ekwoka** [comment, marked as answer]
x-model doesn't really do anything until there is an event on one side or the other.

If it's static like this, just out it in the data, or in an x-init instead.

**Reply by mdmoura** [reply]
I just followed your suggestion and used `data-init` to solve it.

Thank you

**Reply by Kuttoh** [reply]
If anyone is here wondering how to do this, you could use x-init in the input element as follows:

`<input type="hidden" name="subject" x-init="data = { subject: 'my subject'}" x-model="data.subject">`

I had a similar issue and just thought of adding this here for anyone in the future

**Reply by SimoTod** [reply]
FYI @Kuttoh 
x-model has now a `.fill` modifier (it wasn't available when the original question was posted) which does exactly what was needed without hacks.

```
<input type="hidden" name="subject" value="my subject" x-model.fill="data.subject">
 ```

**Reply by Kuttoh** [reply]
Thank you @SimoTod
